src/test/java/com/inventorsoft/junit/service/BookServiceTest.java

```diff
@@ -0,0 +1,220 @@
+        RuntimeException exception = assertThrows(RuntimeException.class,
+                () -> bookService.update(ID, null));
+        assertEquals("Book not found by id: " + ID, exception.getMessage());
+    }
+
+    @Test
+    void update() {
+        UpdateBookRequest updateBookRequest = new UpdateBookRequest();
+        updateBookRequest.setTitle("The Little Prince");
+        updateBookRequest.setDescription("Description");
+        updateBookRequest.setReleaseDate(LocalDate.of(1943, 04, 23));
+        updateBookRequest.setAuthorId(11L);
+
+        Book book1 = new Book();
+        book1.setId(book.getId());
+        book1.setTitle(book.getTitle());
+        book1.setDescription(book.getDescription());
+        book1.setReleaseDate(book.getReleaseDate());
+
+        when(bookRepository.findById(ID)).thenReturn(Optional.of(book));
+        bookMapper.updateEntityFromUpdateRequest(book, updateBookRequest);
+
+        bookService.update(ID, updateBookRequest);
+
+        assertEquals(book1.getId(), book.getId());
+        assertEquals(book1.getTitle(), book.getTitle());
+        assertEquals(book1.getReleaseDate(), book.getReleaseDate());
+        assertEquals(book1.getDescription(), book.getDescription());
+    }
+
+    @Test
+    void changeAuthorForBookShouldThrowRuntimeExceptionWhenBookIdIsNotFound() {
+        when(bookRepository.findById(ID)).thenReturn(Optional.empty());
+        RuntimeException exception = assertThrows(RuntimeException.class,
+                () -> bookService.changeAuthorForBook(ID, null));
+        assertEquals("Book not found by id: " + ID, exception.getMessage());
+    }
+
+    @Test
+    void changeAuthorForBookShouldThrowRuntimeExceptionWhenAuthorIdIsNotFound() {
+        Long newAuthorId = 3L;
+        when(bookRepository.findById(ID)).thenReturn(Optional.of(book));
+        when(authorRepository.findById(newAuthorId)).thenReturn(Optional.empty());
+        RuntimeException exception = assertThrows(RuntimeException.class,
+                () -> bookService.changeAuthorForBook(ID, newAuthorId));
+        assertEquals("Author not found by id: " + newAuthorId, exception.getMessage());
+    }
+
+    @Test
+    void changeAuthorForBook() {
+        Author newAuthor = new Author();
+        Long newAuthorId = 4L;
+        newAuthor.setId(newAuthorId);
+        when(bookRepository.findById(ID)).thenReturn(Optional.of(book));
+        when(authorRepository.findById(newAuthorId)).thenReturn(Optional.of(newAuthor));
+
+        bookService.changeAuthorForBook(ID, newAuthorId);
+
+        assertEquals(newAuthorId,book.getAuthor().getId());
+        verify(bookRepository, times(1)).findById(ID);
+        verify(authorRepository, times(1)).findById(newAuthorId);
+        verify(bookRepository, times(1)).save(book);
+    }
+
+    @Test
+    void deleteById() {
+        bookService.deleteById(ID);
+        verify(bookRepository, times(1)).deleteById(ID);
```

Comment:
`times(1)` could be omitted, it works by default like that
```suggestion
        verify(bookRepository).deleteById(ID);
```